packages/sdk/src/entities/ChainConnection.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  StaticJsonRpcProvider,
+  WebSocketProvider,
+} from '@ethersproject/providers';
+import invariant from 'tiny-invariant';
+
+import {
+  ALCHEMY_WSS_URL,
+  INFURA_RPC_URL,
+  INFURA_WSS_URL,
+} from '../constants/rpcs';
+import { ChainId } from '../enums';
+import { ChainConfigParams } from '../types';
+
+type ChainConnectionParams = {
+  rpcProvider: StaticJsonRpcProvider;
+  wssProvider?: WebSocketProvider;
+};
+
+export class ChainConnection {
+  private static _config: {
```

Comment:
can be converted to Record 🤔

Reply:
`[chainId in ChainId]?` allows optionality while it's not the case with Record afaik